lib/matplotlib/dates.py

```diff
@@ -1314,11 +1334,18 @@ def get_locator(self, dmin, dmax):
         #  bysecond, unused (for microseconds)]
         byranges = [None, 1, 1, 0, 0, 0, None]
 
+        # estimated font ratio since our estimation
+        # is on font size 10
+        font_ratio = (rcParams['font.size'])/10
+
+        # a ratio of 10 date characters per inch is 'estimated'
+        maxwid = rcParams["figure.figsize"][0] * 10
```

Comment:
I think you need to get the width of the axes, not the figure.  I'd also expect 10-pt font characters to be about 72 points-per-inch / 10 points-per-character = 7.2 characters-per-inch wide.

Reply:
Font size generally refers to the height, not the width (i.e., 10pt monospace, sans, serif are not all the same width).

Reply:
Understood, but IMO, the new ticks are *too* spread out.  1) because using the wrong width to work over and 2) because the fontwidth is too wide.

Reply:
I agree with that; almost all the test images that changed appear mostly okay before.